npm/angular-signals/src/mount.ts

```diff
@@ -0,0 +1,525 @@
+  if (typeof component === 'string') {
+    // getTestBed().overrideTemplate is available in v14+
+    // The static TestBed.overrideTemplate is available across versions
+    TestBed.overrideTemplate(WrapperComponent, component)
+
+    return WrapperComponent
+  }
+
+  return component
+}
+
+/**
+ * Creates the ComponentFixture
+ *
+ * @param {Type<T>} component Angular component being mounted
+ * @param {MountConfig<T>} config MountConfig
+
+ * @returns {ComponentFixture<T>} ComponentFixture
+ */
+function setupFixture<T> (
+  component: Type<T>,
+  config: MountConfig<T>,
+): ComponentFixture<T> {
+  const fixture = getTestBed().createComponent(component)
+
+  setupComponent(config, fixture)
+
+  fixture.whenStable().then(() => {
+    fixture.autoDetectChanges(config.autoDetectChanges ?? true)
+  })
+
+  return fixture
+}
+
+// Best known way to currently detect whether or not a function is a signal is if the signal symbol exists.
+// From there, we can take our best guess based on what exists on the object itself
+function isSignal (prop: any): boolean {
+  try {
+    const symbol = Object.getOwnPropertySymbols(prop).find((symbol) => symbol.toString() === 'Symbol(SIGNAL)')
+
+    return !!symbol
+  } catch (e) {
+    // likely a primitive type or something else. We can return false here
+    return false
+  }
+}
+
+// currently not a great way to detect if a function is an InputSignal. When we discover a better way to detect a signal we should incorporate it here
+function isInputSignal (prop: any): boolean {
+  return isSignal(prop) && typeof prop === 'function' && prop['name'] === 'inputValueFn'
+}
+
+// currently not a great way to detect if a function is a Model Signal. Need to improve this
+function isModelSignal (prop: any): boolean {
+  return isSignal(prop) && isWritableSignal(prop) && typeof prop.subscribe === 'function'
+}
+
+// currently not a great way to detect if a function is a Writable Signal. Need to improve this
+function isWritableSignal (prop: any): boolean {
+  return isSignal(prop) && typeof prop === 'function' && typeof prop.set === 'function'
+}
+
+function convertPropertyToSignalIfApplicable (propValue: any, componentValue: any, injector: Injector) {
+  const isComponentValueAnInputSignal = isInputSignal(componentValue)
+  const isComponentValueAModelSignal = isModelSignal(componentValue)
+  let convertedValueIfApplicable = propValue
+
+  // If the component has the property defined as an InputSignal, we need to detect whether a primitive value or not was passed into the component as a prop
+  // and attempt to merge the value in correctly.
+  // We don't want to expose the primitive created signal as it should really be one-way binding from within the component.
+  // However, to make CT testing easier, a user can technically pass in a signal to an input component and assert on the signal itself
+  if (isComponentValueAnInputSignal) {
+    const isPassedInValueNotAFunction = !isFunction(propValue)
```

Comment:
we should likely enhance this to make sure the value passed is in NOT a signal since a signal can really be anything, including objects, arrays, functions, etc